node/cn/api_backend.go

```diff
@@ -96,6 +96,27 @@ func (b *CNAPIBackend) HeaderByNumber(ctx context.Context, blockNr rpc.BlockNumb
 	return header, nil
 }
 
+func (b *CNAPIBackend) HeaderByNumberOrHash(ctx context.Context, blockNrOrHash rpc.BlockNumberOrHash) (*types.Header, error) {
+	if blockNr, ok := blockNrOrHash.Number(); ok {
+		return b.HeaderByNumber(ctx, blockNr)
+	}
+	if hash, ok := blockNrOrHash.Hash(); ok {
+		header, err := b.HeaderByHash(ctx, hash)
+		if err != nil {
+			return nil, err
+		}
+		if header == nil {
+			return nil, fmt.Errorf("header for hash not found")
+		}
+		return header, nil
+	}
+	return nil, fmt.Errorf("invalid arguments; neither block nor hash specified")
+}
+
+func (b *CNAPIBackend) HeaderByHash(ctx context.Context, hash common.Hash) (*types.Header, error) {
+	return b.cn.blockchain.GetHeaderByHash(hash), nil
```

Comment:
`GetHeaderByHash` could return `nil`. How about returning an error to the user instead of returning `nil, nil`?

Reply:
I reflected it. Thank you!